broad/broad/middlewares.py

```diff
@@ -0,0 +1,16 @@
+import os
+
+
+class RandomPayloadMiddleware:
+    def __init__(self, size):
+        self.size = size
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler.settings.getint('SCRAPY_BENCH_RANDOM_PAYLOAD_SIZE'))
+
+    def process_request(self, request, spider):
+        if not self.size or request.body != b'':
+            return
+
+        return request.replace(body=os.urandom(self.size))
```

Comment:
May be instead of body we can use [Rerquest.meta['payload']](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta) here?

Reply:
Hm, but then it would not be sent to the server. I mean this would be OK for benchmarking the queue implementation but you couldn't use it to benchmark the part Scrapy <-> HTTP server.